cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/transform/LogParserTransform.java

```diff
@@ -0,0 +1,183 @@
+  );
+  private static final String LOG_FORMAT_DESCRIPTION = "Log format to parse. Currently only supports S3 and " +
+    "CLF Log format.";
+  private static final String INPUT_TYPE_DESCRIPTION = "Type of the value in the input schema which encodes the " +
+    "log information. Must be either \"string\" or \"bytes\"";
+  private static final String INPUT_NAME_DESCRIPTION = "Name of the value in the input schema which encodes the " +
+    "log information.";
+  private static final Logger LOG = LoggerFactory.getLogger(LogParserTransform.class);
+  private static final Pattern CLF_LOG_PATTERN = Pattern.compile(
+    //   IP                    id    user      date          request     code     size    referrer    user agent
+    "^([\\d.]+|[:][:][\\d]) (\\S+) (\\S+) \\[([^\\]]+)\\] \"([^\"]+)\" (\\d{3}) ([-\"\\d]+) \"([^\"]+)\" \"([^\"]+)\"");
+  private static final Pattern S3_LOG_PATTERN = Pattern.compile(
+    // bucket owner name   time           ip                    req   reqID operation  key     URI
+    "^(\\S+) (\\S+) \\[(\\p{Print}+)\\] ([\\d.]+|[:][:][\\d]) (\\S+) (\\S+) (\\S+) (\\S+) \"([^\"]+)\" " +
+      //  HTTP stat error code  bytes sent     obj size  time    turn time  referrer         user agent
+      "(\\d{3}) (\\p{Print}+) ([-\"\\d]+) ([-\"\\d]+) ([\\d]+) ([-\"\\d]+) \"(\\p{Print}+)\" \"([^\"]+)\" " +
+      //  version id
+      "(\\p{Print}+)");
+  private static final Pattern REQUEST_PAGE_PATTERN = Pattern.compile("(\\S+)\\s(\\S+).*");
+  private static final int[] S3_INDICES = {9, 3, 4, 17};
+  private static final int[] CLF_INDICES = {5, 4, 1, 9};
+  private static final int S3_REGEX_LENGTH = 18;
+  private static final int CLF_REGEX_LENGTH = 9;
+  private final LogParserConfig config;
+
+  public LogParserTransform(LogParserConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter) throws Exception {
+    String log;
+    if ("string".equals(config.inputType)) {
+      log = input.get(config.inputName);
+    } else if ("bytes".equals(config.inputType)) {
+      log = Bytes.toString((ByteBuffer) input.get(config.inputName));
+    } else {
+      LOG.error("Unsupported inputType");
+      return;
+    }
+
+    if (log == null) {
+        LOG.error("Couldn't read schema");
+        return;
+    }
+
+    if ("S3".equals(config.logFormat)) {
+      Matcher logMatcher = S3_LOG_PATTERN.matcher(log);
+      if (!logMatcher.matches() || logMatcher.groupCount() < S3_REGEX_LENGTH) {
+        LOG.error("Couldn't parse log " + log);
+        return;
+      }
+      parseRequest(emitter, logMatcher, S3_INDICES);
+    } else if ("CLF".equals(config.logFormat)) {
+      Matcher logMatcher = CLF_LOG_PATTERN.matcher(log);
+      if (!logMatcher.matches() || logMatcher.groupCount() < CLF_REGEX_LENGTH) {
+        LOG.error("Couldn't parse log " + log);
+        return;
+      }
+      parseRequest(emitter, logMatcher, CLF_INDICES);
+    } else {
```

Comment:
The log format is already parsed into a pattern in the static final constant. Isn't it not possible to do anything further since that requires the actual log?